lib/v2/sjtu/gs.js

```diff
@@ -0,0 +1,55 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+
+module.exports = async (ctx) => {
+    const type = ctx.params.type;
+    const num = ctx.params.num ?? '';
+
+    const rootUrl = 'https://www.gs.sjtu.edu.cn';
+    const currentUrl = `${rootUrl}/announcement/${type}/${num}`;
+    const response = await got({
+        method: 'get',
+        url: currentUrl,
+    });
+
+    const $ = cheerio.load(response.data);
+
+    const list = $('a.announcement-item')
+        .map((_, item) => {
+            item = $(item);
+
+            const day = item.find('.day').text().trim().replace('.', '-');
+            const year = item.find('.month').text().trim();
+
+            return {
+                title: item.find('.title').text().trim(),
+                link: `${item.attr('href').startsWith('http') ? '' : rootUrl}${item.attr('href')}`,
+                pubDate: timezone(parseDate(`${year}-${day}`, 'YYYY-MM-DD'), +8),
+            };
+        })
+        .get();
+
+    const items = await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
+                });
+                const content = cheerio.load(detailResponse.data);
+
+                item.description = (new URL(item.link).hostname === 'mp.weixin.qq.com' ? content('#js_content') : content('.page-content')).html();
```

Comment:
There's a special utility to help you parse articles from `mp.weixin.qq.com` [lib/utils/wechat-mp.js](https://github.com/DIYgod/RSSHub/blob/master/lib/utils/wechat-mp.js)